spec/win32/unit/certstore_spec.rb

```diff
@@ -525,313 +479,202 @@
-      end
-    end
-
-    context "When thumbprint not found" do
-      let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { "invalid_thumbprint" }
-      before(:each) do
-        allow_any_instance_of(certbase).to receive(:get_cert_pem).and_return("")
-      end
-      it "returns empty string" do
-        store = certstore.open(store_name, store_location: store_location)
-        cert_obj = store.get(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: store_name)
-        expect(cert_obj).to be_empty
-      end
-    end
-  end
-
-  describe "#cert_validate" do
-    context "When passing empty certificate store name" do
-      let(:store_name) { "" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      it "raises ArgumentError" do
-        expect { certstore.open(store_name, store_location: store_location) }.to raise_error(ArgumentError, "Empty Certificate Store.")
-      end
-    end
-
-    context "When passing empty thumbprint" do
-      let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { " " }
-      it "raises ArgumentError" do
-        store = certstore.open(store_name, store_location: store_location)
-        expect { store.valid?(thumbprint) }.to raise_error(ArgumentError, "Invalid certificate thumbprint.")
-      end
-    end
-
-    context "When passing thumbprint is nil" do
-      let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { nil }
-      it "raises ArgumentError" do
-        store = certstore.open(store_name, store_location: store_location)
-        expect { store.valid?(thumbprint) }.to raise_error(ArgumentError, "Invalid certificate thumbprint.")
-      end
-    end
-
-    context "When passing invalid thumbprint" do
-      let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { "b1bc968bd4f49d622aa89a81f2150152a41d829c" }
-      before(:each) do
-        allow_any_instance_of(certbase).to receive(:get_cert_pem).and_return("")
-      end
-      it "returns false" do
-        store = certstore.open(store_name, store_location: store_location)
-        expect(store.valid?(thumbprint)).to eql(false)
-      end
     end
 
-    context "When passing valid certificate's thumbprint" do
+    context "When passing a valid thumbprint with spaces or colons to the LocalMachine store" do
       let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { "b1bc968bd4f49d622aa89a81f2150152a41d829909c" }
-      let(:cert_pem) { File.read('.\spec\win32\assets\GlobalSignRootCA.pem') }
+      let(:thumbprint) { "b1 bc 96 8b d4 f4 9d 62 2a a8 9a 81 f2 15 01 52 a4 1d 82 9c" }
+      let(:thumbprint2) { "b1:bc:96:8b:d4:f4:9d:62:2a:a8:9a:81:f2:15:01:52:a4:1d:82:9c" }
```

Comment:
are the changes to the thumbprint format here also a breaking change to the API?

Reply:
I undid those to be more permissive about thumbprint format.